.github/workflows/check-rust.yml

```diff
@@ -0,0 +1,136 @@
+name: Rust build, benchmarks, and tests
+
+concurrency:
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+##
+on:
+  ##
+  # Run automatically for any push that changes Rust file(s)
+  push:
+    branches:
+      - main
+      - $default-branch
+
+    paths:
+      - "**.rs"
+
+  ##
+  # Run automatically for PRs against default/main branch if Rust files change
+  pull_request:
+    branches:
+      - main
+      - $default-branch
+
+    paths:
+      - "**.rs"
+
+  ## Allow running workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      verbose:
+        description: "Output more information when triggered manually"
+        required: false
+        default: ""
+
+##
+env:
+  CARGO_TERM_COLOR: always
+  VERBOSE: ${{ github.events.input.verbose }}
+
+##
+#
+jobs:
+  check:
+    name: Tests targeting ${{ matrix.rust-target }} for OS ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    ##
+    # Define multiple targets for builds and tests
+    strategy:
+      matrix:
+        rust-branch:
+          - stable
+
+        rust-target:
+          - x86_64-unknown-linux-gnu
+          # - x86_64-apple-darwin
+
+        os:
+          - ubuntu-latest
+          # - macos-latest
+
+        include:
+          - os: ubuntu-latest
+          # - os: macos-latest
+
+    ##
+    env:
+      RELEASE: false
+      RELEASE_NAME: development
+      RUSTFLAGS: -A warnings
+      RUSTV: ${{ matrix.rust-branch }}
+      RUST_BACKTRACE: full
+      RUST_BIN_DIR: target/${{ matrix.rust-target }}
+      SKIP_WASM_BUILD: 1
+      TARGET: ${{ matrix.rust-target }}
+
+    ##
+    steps:
+      - name: Check-out repository under $GITHUB_WORKSPACE
+        uses: actions/checkout@v2
+
+      - name: Set RELEASE mode
+        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
```

Comment:
do we have a staging branch here?? I am not able to find it
We don't use a staging branch in this repository, is that correct?

Reply:
:facepalm: oof, good catch!

And checking a [blog][1] entry from a few years ago, it seems we can now utilize `$default-branch` to avoid future churn.  I'll get this fixed in a few minutes for sure

[1]: https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/